src/Spice86.Core/Emulator/Devices/Timer/Pit8254Counter.cs

```diff
@@ -187,17 +247,102 @@ public void WriteReloadValueByte(byte value) {
 
     /// <summary>
     /// Checks whether the counter is activated and if so, decrements the ticks.
+    /// Implements the behavior defined in Intel 8254 Datasheet section 3.3 "Mode Definitions"
     /// </summary>
     /// <returns>Whether the activation was processed.</returns>
     public bool ProcessActivation() {
         if (Activator.IsActivated) {
+            // For certain modes, counting is inhibited when gate is low
+            // "When GATE is low, counting is inhibited." - Intel 8254 datasheet
+            bool shouldCount = true;
+            switch (CurrentPitMode) {
```

Comment:
Extract the computation of should count to a function? This could make code intent clearer

Reply:
Done.